stl/inc/xutility

```diff
@@ -5563,17 +5563,42 @@ _NODISCARD constexpr auto lexicographical_compare_three_way(
 }
 #endif // __cpp_lib_concepts
 
+template <class _Ty, class _Elem>
+struct _Vector_alg_in_find_is_safe_object_pointers : false_type {};
+template <class _Ty1, class _Ty2>
+struct _Vector_alg_in_find_is_safe_object_pointers<_Ty1*, _Ty2*>
+    : conjunction<
+          // _Ty1* is an object pointer type
+          disjunction<is_object<_Ty1>, is_void<_Ty1>>,
+          // _Ty2* is an object pointer type
+          disjunction<is_object<_Ty2>, is_void<_Ty2>>,
+          // either _Ty1 is the same as _Ty2 (ignoring cv-qualifiers), or one of the two is void
+          disjunction<is_same<remove_cv_t<_Ty1>, remove_cv_t<_Ty2>>, is_void<_Ty1>, is_void<_Ty2>>> {};
+
+// Can we activate the vector algorithms for find/count?
 template <class _Iter, class _Ty, class _Elem = _Iter_value_t<_Iter>>
-_INLINE_VAR constexpr bool _Vector_alg_in_find_is_safe = // Can we activate the vector algorithms for find/count?
-    _Iterator_is_contiguous<_Iter> // The iterator must be contiguous so we can get raw pointers.
-    && !_Iterator_is_volatile<_Iter> // The iterator must not be volatile.
-    && disjunction_v< // And one of the following conditions must be met:
+_INLINE_VAR constexpr bool _Vector_alg_in_find_is_safe =
+    // The iterator must be contiguous so we can get raw pointers.
+    _Iterator_is_contiguous<_Iter>
+    // The iterator must not be volatile.
+    && !_Iterator_is_volatile<_Iter>
+    // And one of the following conditions must be met:
+    && disjunction_v<
 #ifdef __cpp_lib_byte
-        conjunction<is_same<_Ty, byte>, is_same<_Elem, byte>>, // We're finding a std::byte in a range of std::byte.
+        // We're finding a std::byte in a range of std::byte.
+        conjunction<is_same<_Ty, byte>, is_same<_Elem, byte>>,
 #endif // __cpp_lib_byte
-        conjunction<is_integral<_Ty>, is_integral<_Elem>>, // We're finding an integer in a range of integers.
-        // The integer types can be different, which requires careful handling.
-        conjunction<is_pointer<_Ty>, is_same<_Ty, _Elem>>>; // We're finding a U* in a range of U* (identical types).
+       // We're finding an integer in a range of integers.
+       // The integer types can be different, which requires careful handling.
+        conjunction<is_integral<_Ty>, is_integral<_Elem>>,
+        // We're finding a pointer in a range of pointers of the same type (U* or U(*)(Ts...)).
+        conjunction<is_pointer<_Ty>, is_same<_Ty, _Elem>>,
+        // We're finding a nullptr in a range of pointers (U* or U(*)(Ts...)).
+        conjunction<is_same<_Ty, nullptr_t>, is_pointer<_Elem>>,
+        // We're finding an object pointer in a range of object pointers, and:
+        // - One of the pointers is a cv void*.
+        // - One of the pointers is a cv1 U* and the other is a cv2 U*.

```

Comment:
I find it confusing that we first use "pointer(s)" to refer to values, and then use "pointers" to refer to types.
```suggestion
        // - One of the pointer types is cv void*, or
        // - One of the pointer types is cv1 U* and the other is cv2 U*.
```